### ISSUE TITLE
Eliminate Compilers.Toolset 4.5.0 pre-built in source-build

### ISSUE THREAD

**Issue by oleksandr-didyk** [dotnet/roslyn-analyzers#6618]
Part of https://github.com/dotnet/roslyn-analyzers/issues/6606

`Microsoft.Net.Compilers.Toolset: 4.5.0` is a [direct dependency](https://github.com/dotnet/roslyn-analyzers/blob/main/eng/Version.Details.xml#L15) that is being flagged as a pre-built in source-build. 

Ideal scenario for resolving the pre-built would be to bump it to the latest version, which at the time of writing would cause a build-time exception:
```
/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataOperationVisitor.cs(462,101): error CS8601: Possible null reference assignment. [/src/NetAnalyzers/Core/Microsoft.CodeAnalysis.NetAnalyzers.csproj]
```
Furthermore, I am not sure if using latest version would even make sense in this case.

In cases where a historic version is required, the source-build team would take the package and create a reference assembly form it an utilize it in the source-build. However, due to the nature of the package we currently cannot create a reference pack from it.

It would be great if we could get confirmation about this package version - does it need to be at `4.5.0` or can it use a different version if required?

If not, we could add an exclusion rule for it as was done in the past for [`roslyn`](https://github.com/dotnet/roslyn/pull/67645/files#diff-f88ce4026366ce74fb1a981785c542757617abf4393a33ca34465a5e9a0dd343R13) (@MichaelSimons can correct me if this is not possible)

**Comment by sharwell** [comment]
@oleksandr-didyk Can you clarify how reference assemblies apply here? It's my understanding that pre-builts are only a problem if we need to execute code against it.

**Comment by oleksandr-didyk** [comment]
> @oleksandr-didyk Oleksandr Didyk FTE Can you clarify how reference assemblies apply here? It's my understanding that pre-builts are only a problem if we need to execute code against it.

In situations where a repository has a dependency on a pinned historic stable version of a given library, the source-build team would try to utilize a reference package for it. This package is essentially a reference assembly with some differences iirc (a DLL containing only the API surface of a given package without any implementation). Due to its nature, the reference package can be compiled against, but will fail if the build tries to execute it.

`Compilers.Toolset: 4.5.0` satisfies the _pinned historic stable version_ requirement, but the package itself is not a simple library but some special package. 

I am not that experienced with `Compilers.Toolset` and based my options with dealing with it off work performed by @mmitche. Maybe he would be of better help here

**Comment by mmitche** [comment]
Taking a look.

**Comment by mmitche** [comment]
For Microsoft.NET.Compilers.Toolset in this repo, it appears that the repo is only using this to pin the version of roslyn it compiles with. It has nothing to do with the surface area that RA is building functionality against.

My gut says that this repo should simply use the SDK-provided compiler. Get rid of the direct dependency in `VersionDetails.xml`, the `MicrosoftNetCompilersToolsetVersion` property, and `<UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>` (defaults to false).

Fix the compile errors that result.

**Comment by mmitche** [comment]
After looking at this a bit further, unless we see a bunch of compile time differences between the live-built version of roslyn used in SB, and the 4.5.0 stable version that this repo uses, we should keep on the current course and just baseline this dependency.

**Comment by oleksandr-didyk** [comment]
This issue was resolved by https://github.com/dotnet/roslyn-analyzers/issues/6606, just wasn't mentioned in the PR.

Closing
